app/views/main/segment_page.erb

```diff
@@ -0,0 +1,10 @@
+
+<center>
+<h1>Segmentation!</h1>
+<%=
+input = params[:string]
+x = StringSegmenter.new
+x.run_program(input)
+%>
```

Comment:
Too many lines of Ruby in the view file. Controllers are supposed to prepare information for the view. Remember that if you define a variable in the controller and name it with an `@` in front of it, you can refer to it in the view.